src/snowflake/snowpark/session.py

```diff
@@ -1710,7 +1797,9 @@ def _get_available_versions_for_packages(
     @property
     def query_tag(self) -> Optional[str]:
         """
-        The query tag for this session.
+        The query tag for this session. When session object is shared between multiple
+        threads, each thread will have its own query tag and will need to be initialized
+        individually.
```

Comment:
this can't be done. query tag is common for whole session since we run an alter session set.